test.js

```diff
@@ -37,6 +42,15 @@ test('should use the defaultDnsType where provided', (t) => {
   data.forEach(d => t.is(toMultiaddr(d[1], d[2]).toString(), d[0], `Converts ${d[1]} to ${d[0]} with opts ${d[2]}`))
 })
 
-test('should throw for unsupported protocol', (t) => {
-  t.throws(() => toMultiaddr('quic://'))
+test('should throw for on invalid url', (t) => {
+  t.throws(() => {
+    toMultiaddr('whoosh.fast')
+  }, /Invalid URL/)
+})
+
+test('should throw for unknown protocol', (t) => {
+  t.throws(() => {
+    // NOTE: `data` is a valid uri protocl but isn't a valid multiaddr protocol yet
```

Comment:
```suggestion
    // NOTE: `data` is a valid uri protocol but isn't a valid multiaddr protocol yet
```

This is a cool edge case! :)